File: frontend/src/services/dhan_credentials.js
/**
 * Dhan Credentials Service
 *
 * API calls for Dhan static token credential management (Tier 3).
 */
import api from './api'

export async function getCredentials() {
  const response = await api.get('/api/dhan-credentials/credentials')
  return response.data
}

export async function storeCredentials(credentials) {
  const response = await api.post('/api/dhan-credentials/credentials', credentials)
  return response.data
}

export async function deleteCredentials() {
  const response = await api.delete('/api/dhan-credentials/credentials')
  return response.data
}

export default { getCredentials, storeCredentials, deleteCredentials }

File: backend/app/schemas/dhan_credentials.py
"""
Dhan Credentials Pydantic Schemas

Schemas for Dhan static token credential management (Tier 3).
"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class DhanCredentialsCreate(BaseModel):
    """Request schema for storing Dhan credentials."""
    client_id: str = Field(..., min_length=1, max_length=50, description="Dhan Client ID")
    access_token: str = Field(..., min_length=1, description="Dhan Access Token")


class DhanCredentialsResponse(BaseModel):
    """Response schema for Dhan credentials status (never returns actual token)."""
    has_credentials: bool
    client_id: Optional[str] = None
    is_active: bool = False
    last_auth_at: Optional[datetime] = None
    last_error: Optional[str] = None

    class Config:
        from_attributes = True
